api/base/filters.py

```diff
@@ -66,6 +69,33 @@ def filter_queryset(self, request, queryset, view):
             return queryset.sort(*ordering)
         return queryset
 
+    def check_serializer_fields(self, queryset, query_params, view, ordering):
+        serializer_class = view.serializer_class
+        if not queryset:
+            return []
+        sorting_params = []
+        for i, field in enumerate(query_params.getlist(self.ordering_param)):
+            preserve_order = ''
+            # If the field is already a source, it will be returned in the ordering list
+            if field in ordering:
+                sorting_params.append(field)
+                continue
+            # The field may have a '-' before the name for descending order. This needs to be preserved
+            if field[0] == '-':
+                preserve_order = '-'
+                field = field[1:]
```

Comment:
This is nit picky, but python has a builtin func for this `.lstrip('-')`, also `preserve_order` isn't really named well might want to just make that a bool.